terrain/instancing/voxel_instancer.cpp

```diff
@@ -714,6 +714,7 @@ void VoxelInstancer::update_block_from_transforms(int block_index, ArraySlice<co
 		block->multimesh_instance.set_world(world);
 		block->multimesh_instance.set_transform(block_transform);
 		block->multimesh_instance.set_material_override(item->get_material_override());
+		block->multimesh_instance.set_cast_shadows_setting(item->get_cast_shadows_setting());
```

Comment:
If you want it to update in editor, you can also add this to https://github.com/Zylann/godot_voxel/blob/ef5ec4b059117f0fded1f90b31692f33174b15f8/terrain/instancing/voxel_instancer.cpp#L438